### PR TITLE
Vue-electron

### PR DESCRIPTION
**Pull request by NathanBland**
This is the most complete UI for `garden` yet, and as far as I know the only usable interface.

### CODE REVIEW

File: app/src/main.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  routes
+})
+
+router.beforeEach((to, from, next) => {
+  // to and from are both route objects
+  // console.log('going to:', to)
+  next()
+})
+
+const store = new Vuex.Store({
```
Comment (crodeheaver) [inline app/src/main.js:31]: I'm assuming you'll have a bit of state as time goes on, so I'd probably move this out to it's own file. Maybe even make a module or two.

File: app/src/main.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import 'bulma/css/bulma.css'
+import 'font-awesome/css/font-awesome.css'
+
+Vue.use(Electron)
+Vue.use(Resource)
+Vue.use(Vuex)
+Vue.use(Router)
+Vue.config.debug = true
+Vue.http.options.root = 'https://garden-api.herokuapp.com/api'
```
Comment (crodeheaver) [inline app/src/main.js:18]: Will webpack/vue allow you to put a environment variable here? Or maybe the vuex could store it as a sort of a config type module/storage?

**Review by crodeheaver — COMMENTED** [review]
Not that I know vue or electron very well, but it all looks ok to me.

File: app/src/main.js (hunk 1 above)
Comment (NathanBland) [reply app/src/main.js:31]: The only state I plan to track with this globally is the authentication... 

Everything else is tracked by component level data.

However, you may be right, that at some point, that would have an advantage.

File: app/src/main.js (hunk 2 above)
Comment (NathanBland) [reply app/src/main.js:18]: I don't know why not, for the second, but the first doesn't work with the current `npm` scripts. I've tried on other projects.

**Comment by crodeheaver** [comment]
I'm fine merging. You can get a second opinion if you want, or I'll just merge now.

**Comment by NathanBland** [comment]
:shipit: (shipIt)

**Pull request merged**